osu.Game/Overlays/OverlayHeader.cs

```diff
@@ -15,9 +15,18 @@ public abstract class OverlayHeader : Container
     {
         private readonly Box titleBackground;
         private readonly ScreenTitle title;
+        private readonly Container content;
 
         protected readonly FillFlowContainer HeaderInfo;
 
+        public const float CONTENT_X_MARGIN = 50;
+
+        public virtual float HorizontalMargin
```

Comment:
Should be `protected`, get-only and used directly in the ctor. Basically make `CONTENT_X_MARGIN` into a protected get-only property.

Reply:
Sorry, I don't think I understood this properly, could you elaborate? 

I don't agree that `CONTENT_X_MARGIN` should be made into anything `protected` since it might be used by other components outside the `OverlayHeader` or any classes deriving from it (see current `UserProfileOverlay.CONTENT_X_MARGIN and how it's used by other parts of the overlay). Unless this can be achieved in a different way?

We also need the header to let us customize this value in case an overlay needs a different one - like the `UserProfileOverlay` does in this PR - that's why I made `HorizontalMargin` like this. I do indeed feel like there's a better way to do that, but I don't have any concrete ideas right now.

Reply:
OK, those consts can stay for now I suppose, although they feel super wrong. Here I mainly have an issue with exposition as public. IMO would be better to invert flow:

```csharp
protected virtual float HorizontalPadding => CONTENT_X_MARGIN;
```

and then instead at `content` creation pass `new MarginPadding { Horizontal = HorizontalPadding }`.

Reply:
Changed, please check if it's better now.